src/common/utils/time/time.ts

```diff
@@ -175,68 +148,41 @@ export function formatTimeElapsed(date: Date, timezone: Timezone): string {
 }
 
 export function formatDateTime(date: Date, timezone: Timezone): string {
-  return getMoment(date, timezone).format(FORMAT_FULL_DATE_TIME);
-}
-
-export function formatDate(date: Date, timezone: Timezone): string {
-  return getMoment(date, timezone).format(FORMAT_DATE);
+  return getMoment(date, timezone).format(FULL_FORMAT);
 }
 
-export function formatTime(date: Date, timezone: Timezone): string {
-  return getMoment(date, timezone).format(FORMAT_TIME);
+export function formatISODate(date: Date, timezone: Timezone): string {
+  return getMoment(date, timezone).format(ISO_FORMAT_DATE);
 }
 
-function pad(input: number): string {
-  if (input < 10) return `0${input}`;
-  return String(input);
+export function formatISOTime(date: Date, timezone: Timezone): string {
+  return getMoment(date, timezone).format(ISO_FORMAT_TIME);
 }
 
-export function formatTimeBasedOnGranularity(range: DateRange, granularity: Duration, timezone: Timezone, locale: Locale): string {
-  const wallTimeStart = getMoment(range.start, timezone);
-  const year = wallTimeStart.year();
-  const month = wallTimeStart.month();
-  const day = wallTimeStart.date();
-  const hour = wallTimeStart.hour();
-  const minute = wallTimeStart.minute();
-  const second = wallTimeStart.second();
+const ISO_DATE_DISALLOWED = /[^\d-]/g;
 
-  const monthString = locale.shortMonths[month];
-  const hourToTwelve = hour % 12 === 0 ? 12 : hour % 12;
-  const amPm = (hour / 12) >= 1 ? "pm" : "am";
+export function trimISODate(date: string): string {
+  return date.replace(ISO_DATE_DISALLOWED, "");
+}
 
-  const granularityString = granularity.toJS();
-  const unit = granularityString.substring(granularityString.length - 1);
+const ISO_DATE_TEST = /^\d\d\d\d-\d\d-\d\d$/;
 
-  switch (unit) {
-    case "S":
-      return `${monthString} ${day}, ${pad(hour)}:${pad(minute)}:${pad(second)}`;
-    case "M":
-      const prefix = granularityString.substring(0, 2);
-      return prefix === "PT" ? `${monthString} ${day}, ${hourToTwelve}:${pad(minute)}${amPm}` : `${monthString}, ${year}`;
-    case "H":
-      return `${monthString} ${day}, ${year}, ${hourToTwelve}${amPm}`;
-    case "D":
-      return `${monthString} ${day}, ${year}`;
-    case "W":
-      return `${formatTimeRange(range, timezone, DisplayYear.ALWAYS)}`;
-    default:
-      return wallTimeStart.format(FORMAT_ISO_WITHOUT_TIMEZONE);
-  }
+export function validateISODate(date: string): boolean {
+  return ISO_DATE_TEST.test(date);
 }
 
-export function formatGranularity(granularity: string): string {
-  return granularity.replace(/^PT?/, "");
-}
+const ISO_TIME_DISALLOWED = /[^\d:]/g;
 
-export function maybeFullyDefinedDate(date: string): boolean {
-  return date.length === FORMAT_DATE.length;
+export function trimISOTime(time: string): string {
```

Comment:
I would change the verb to *prepare* or *normalize* as the name is a bit misleading now